File: project/jimmer-apt/src/main/java/org/babyfish/jimmer/apt/JacksonTypes.java
package org.babyfish.jimmer.apt;

import com.squareup.javapoet.ClassName;

public class JacksonTypes {

    public final ClassName jsonIgnore;

    public final ClassName jsonValue;

    public final ClassName jsonPropertyOrder;

    public final ClassName jsonFormat;

    public final ClassName jsonSerializer;

    public final ClassName jsonSerialize;

    public final ClassName jsonDeserialize;

    public final ClassName jsonPojoBuilder;

    public final ClassName jsonNaming;

    public final ClassName jsonGenerator;

    public final ClassName serializerProvider;

    public JacksonTypes(
            ClassName jsonIgnore,
            ClassName jsonValue,
            ClassName jsonPropertyOrder,
            ClassName jsonFormat,
            ClassName jsonSerializer,
            ClassName jsonSerialize,
            ClassName jsonDeserialize,
            ClassName jsonPojoBuilder,
            ClassName jsonNaming,
            ClassName jsonGenerator,
            ClassName serializerProvider
    ) {
        this.jsonIgnore = jsonIgnore;
        this.jsonValue = jsonValue;
        this.jsonPropertyOrder = jsonPropertyOrder;
        this.jsonFormat = jsonFormat;
        this.jsonSerializer = jsonSerializer;
        this.jsonSerialize = jsonSerialize;
        this.jsonDeserialize = jsonDeserialize;
        this.jsonPojoBuilder = jsonPojoBuilder;
        this.jsonNaming = jsonNaming;
        this.jsonGenerator = jsonGenerator;
        this.serializerProvider = serializerProvider;
    }
}

File: project/jimmer-ksp/src/main/kotlin/org/babyfish/jimmer/ksp/JacksonTypes.kt
package org.babyfish.jimmer.ksp

import com.squareup.kotlinpoet.ClassName

class JacksonTypes(
    val jsonIgnore: ClassName,
    val jsonValue: ClassName,
    val jsonFormat: ClassName,
    val jsonProperty: ClassName,
    val jsonPropertyOrder: ClassName,
    val jsonCreator: ClassName,
    val jsonSerializer: ClassName,
    val jsonSerialize: ClassName,
    val jsonDeserialize: ClassName,
    val jsonPojoBuilder: ClassName,
    val jsonNaming: ClassName,
    val jsonGenerator: ClassName,
    val serializeProvider: ClassName
)